simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/evse/states/ChargingState.java

```diff
@@ -29,20 +33,23 @@ public String getStateName() {
     }
 
     @Override
-    public void onPlug(int evseId, int connectorId) {
-        // NOP
+    public CompletableFuture<UserMessageResult> onPlug(int evseId, int connectorId) {
+        return CompletableFuture.completedFuture(UserMessageResult.NOT_EXECUTED);
     }
 
     @Override
-    public void onAuthorize(int evseId, String tokenId) {
+    public CompletableFuture<UserMessageResult> onAuthorize(int evseId, String tokenId) {
         log.info("in authorizeToken {}", tokenId);
 
+        CompletableFuture<UserMessageResult> future = new CompletableFuture<>();
         StationMessageSender stationMessageSender = stateManager.getStationMessageSender();
         stationMessageSender.sendAuthorizeAndSubscribe(tokenId, singletonList(evseId),
-                (request, response) -> handleAuthorizeResponse(evseId, tokenId, response));
+                (request, response) -> handleAuthorizeResponse(evseId, tokenId, response, future));
```

Comment:
it is a super bad idea to mutate future object inside the method, 
can you make it return future instead?

Reply:
or maybe result, can construct future then?